### PR TITLE
Mass cleanup of snippet chooser modal code

### PR DESCRIPTION
**Pull request by gasman**
Started out with the aim of phasing out ajax_pagination_nav.html - ended up going down quite a bit more of a rabbithole :-)

@kaedroho I've assigned you to review, as it encroaches on the locale filter code.

### CODE REVIEW

File: wagtail/snippets/urls.py [hunk 1]
```diff
@@ -7,7 +7,6 @@
 urlpatterns = [
     path('', snippets.index, name='index'),
 
-    path('choose/', chooser.choose, name='choose_generic'),
```
Comment (gasman) [inline wagtail/snippets/urls.py:10]: To clarify, choose_generic was never a usable URL route in its own right - it only served to generate the partial URL to pass to JS (where the model parameter would then get tacked on).

File: wagtail/admin/templatetags/wagtailadmin_tags.py [hunk 2]
```diff
@@ -716,3 +717,19 @@ def menu_props(context):
 @register.simple_tag
 def get_comments_enabled():
     return getattr(settings, 'WAGTAILADMIN_COMMENTS_ENABLED', True)
+
+
+@register.simple_tag
+def resolve_url_if_not_absolute(url):
```
Comment (kaedroho) [inline wagtail/admin/templatetags/wagtailadmin_tags.py:723]: Django has a similar built-in function https://github.com/django/django/blob/4b6208ffdd93d5d379d247ec094d567ef208effd/django/shortcuts.py#L102

Have you considered using this?

**Review by kaedroho — APPROVED** [review]
This looks good to me, thanks!

File: wagtail/admin/templatetags/wagtailadmin_tags.py (hunk 2 above)
Comment (gasman) [reply wagtail/admin/templatetags/wagtailadmin_tags.py:723]: Ah, nice! Will update to use that...

**Pull request merged**